packages/rspack/src/utils/create-compiler.ts

```diff
@@ -15,7 +19,15 @@ export async function createCompiler(
     userDefinedConfig = await userDefinedConfig;
   }
 
-  const config = await userDefinedConfig({}, { options, context });
+  const config: RspackOptions = await userDefinedConfig(
+    {},
+    { options, context }
+  );
+
+  config.output = {
+    ...config.output,
+    filename: options.outputFileName,
```

Comment:
I think `config.entry` needs to be set rather than `config.output.filename` because the latter controls the name of chunks and other entry points.

something like: 

```
const mainEntry = options.outputFileName
  ? path.parse(options.outputFileName).name // remove extension
  : 'main'; // default is `main.js`
config.entry[mainEntry] = options.main;
```

Reply:
Hm, I thought that we were giving the option to the user to specify the name of the main output file like we do here (https://nx.dev/packages/node/executors/webpack#outputfilename), as described here: https://www.rspack.dev/config/output.html#outputfilename, but maybe I misunderstood?